src/OpenFeature.Providers.GOFeatureFlag/evaluator/IEvaluator.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Threading.Tasks;
+using OpenFeature.Model;
+
+namespace OpenFeature.Providers.GOFeatureFlag.Evaluator;
+
+/// <summary>
+///     IEvaluator is an interface that represents the evaluation of a feature flag.
+///     It can have multiple implementations: Remote or InProcess.
+/// </summary>
+public interface IEvaluator : IAsyncDisposable
+{
+    /// <summary>
+    ///     Initialize the evaluator.
+    /// </summary>
+    public Task InitializeAsync();
+
+    /// <summary>
+    ///     Evaluate an Object flag.
+    /// </summary>
+    /// <param name="flagKey">name of the feature flag</param>
+    /// <param name="defaultValue">default value</param>
+    /// <param name="evaluationContext">evaluation context of the evaluation</param>
+    /// <returns>An ResolutionDetails response</returns>
+    public Task<ResolutionDetails<Value>> EvaluateAsync(string flagKey, Value defaultValue,
```

Comment:
I wonder if we should have the `CancellationToken` being passed here so it can be used in case the client needs to cancel the flow execution.

Reply:
Good catch, I've introduced it.